dts/bindings/reserved-memory/reserved-memory.yaml

```diff
@@ -0,0 +1,26 @@
+description: Reserved memory
+
+compatible: "reserved-memory"
```

Comment:
It's kind of a shame that we need a compatible here. Maybe it's worth it to add edtlib support for accepting a binding for a node at a particular path in the tree, kind of like how inferred bindings support works, but with an explicit binding instead of an inferred one. Would that be useful to you, or is the inconvenience of having to type a compatible in the DTS not a big deal for you?

Reply:
I think there could be some value in this.

Reply:
yeah, I think that we can leverage that compatible in the future somehow. For example having a `reserved-memory` driver having to do some work on the regions.

Reply:
@carlocaione -- it sounds like you're saying you prefer to keep the compatible?